src/svc_ioq.c

```diff
@@ -353,7 +353,7 @@ void svc_ioq_write(SVCXPRT *xprt)
 				"%s: %p fd %d About to destroy - rc = %d",
 				__func__, xprt, xprt->xp_fd, rc);
 			SVC_DESTROY(xprt);
-			break;
+			/* Continue to clenaup all requests in queue */
```

Comment:
So, this definitely won't work on it's own, it will just loop forever (possibly accessing invalid memory) since we haven't updated "have".   We won't be able to actually process any more events, since the xprt has been destroyed, so we won't (and can't) even run svc_ioq_flush(). 

Actually, it's worse, I think.  With this change, we'll loop back to the beginning, reset rc=0, then proceed to fall into the success path on the same "have".  But that on should have failed.

I'm not sure there's a good way to clean up the writeq in this codepath, at least without some re-organization.

Reply:
> So, this definitely won't work on it's own, it will just loop forever (possibly accessing invalid memory) since we haven't updated "have". We won't be able to actually process any more events, since the xprt has been destroyed, so we won't (and can't) even run svc_ioq_flush().
> 
> Actually, it's worse, I think. With this change, we'll loop back to the beginning, reset rc=0, then proceed to fall into the success path on the same "have". But that on should have failed.

we will loop back to the beginning with SVC_DESTROY, so we should not end up in svc_ioq_flushv since we have check for !(xprt->xp_flags & SVC_XPRT_FLAG_DESTROYED).
So we will go ahead with rc=0, process existing xioq as success do XDR_DESTROY to cleanup the xioq, get next "have", it will do same in loop without calling svc_ioq_flushv, which should cleanup all the responses queues.